eventmesh-connector-plugin/eventmesh-connector-standalone/src/main/java/org/apache/eventmesh/connector/standalone/broker/task/SubScribeTask.java

```diff
@@ -0,0 +1,109 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.connector.standalone.broker.task;
+
+import io.openmessaging.api.AsyncMessageListener;
+import io.openmessaging.api.Message;
+import org.apache.eventmesh.api.EventMeshAction;
+import org.apache.eventmesh.api.EventMeshAsyncConsumeContext;
+import org.apache.eventmesh.connector.standalone.broker.StandaloneBroker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SubScribeTask implements Runnable {
+
+    private String               topicName;
+    private StandaloneBroker     standaloneBroker;
+    private AsyncMessageListener listener;
+    private volatile boolean isRunning;
+
+    private AtomicInteger offset;
+
+    private final Logger logger = LoggerFactory.getLogger(SubScribeTask.class);
+
+    public SubScribeTask(String topicName,
+                         StandaloneBroker standaloneBroker,
+                         AsyncMessageListener listener) {
+        this.topicName = topicName;
+        this.standaloneBroker = standaloneBroker;
+        this.listener = listener;
+        this.isRunning = true;
+    }
+
+    @Override
+    public void run() {
+        while (isRunning) {
+            try {
+                logger.debug("execute subscribe task, topic: {}, offset: {}", topicName, offset);
+                if (offset == null) {
+                    Message message = standaloneBroker.getMessage(topicName);
+                    if (message != null) {
+                        offset = new AtomicInteger((int) message.getOffset());
+                    }
+                }
+                if (offset != null) {
+                    Message message = standaloneBroker.getMessage(topicName, offset.get());
+                    if (message != null) {
+                        EventMeshAsyncConsumeContext consumeContext = new EventMeshAsyncConsumeContext() {
+                            @Override
+                            public void commit(EventMeshAction action) {
+                                switch (action) {
+                                    case CommitMessage:
+                                        // update offset
+                                        offset.incrementAndGet();
+                                        logger.info("message commit, topic: {}, current offset:{}", topicName, offset.get());
+                                        break;
+                                    case ReconsumeLater:
+                                        // don't update offset
+                                        break;
+                                    case ManualAck:
+                                        // update offset
+                                        offset.incrementAndGet();
```

Comment:
Here is the ManualAck, shouldn't update the offset after commit the message, there is an acktask in runtime, only client call this task, the offset will be increment

Reply:
OK, please take a look.